src/lib.rs

```diff
@@ -90,6 +90,34 @@ fn get_num_cpus() -> usize {
     }
 }
 
+/// Returns the number of physical CPUs of the current machine.
+#[inline]
+pub fn get_physical() -> usize {
+    get_physical_num_cpus()
+}
+
+#[cfg(target_os="macos")]
+fn get_physical_num_cpus() -> usize {
+    use std::ffi::CString;
+    use libc::size_t;
+    use libc::sysctlbyname;
+    use std::ptr;
+    use libc::c_void;
+
+    unsafe {
+        let name = CString::new("hw.physicalcpu").unwrap().as_ptr();
```

Comment:
This could be a stack array:

```rust
let name = [104u8, 119, 46, 112, 104, 121, 115, 105, 99, 97, 108, 99, 112, 117, 0];
let name_ptr = name.as_ptr();
```